hibernate-core/src/main/java/org/hibernate/event/internal/DefaultPostLoadEventListener.java

```diff
@@ -51,7 +51,7 @@ public void onPostLoad(PostLoadEvent event) {
 		switch ( entry.getLockMode() ) {
 			case PESSIMISTIC_FORCE_INCREMENT:
 				final Object nextVersion = entry.getPersister()
-						.forceVersionIncrement( entry.getId(), entry.getVersion(), session );
+						.forceVersionIncrement( entry.getId(), entry.getVersion(), false, session );
```

Comment:
Wouldn't it maybe be better to also call `executeBatch` in `SessionImpl#afterOperation` and `StatelessSessionImpl#afterOperation` to avoid disabling batching here since it could be beneficial?
If disabling batching here is not an option, we should probably also disable batching in
`org.hibernate.loader.ast.internal.LoaderHelper#upgradeLock` and `org.hibernate.dialect.lock.PessimisticForceIncrementLockingStrategy#lock` that both seem to want to `forceVersionIncrement` based on this lock mode as well.

Reply:
I was not sure disabling batch was correct  , I think all depends when the query have to be executed, if I'm not wrong with batch it is executed after a flush or when the transaction is closed and I think in this cases is wrong.

Reply:
the select query triggers the batch containing the insert for `entity1` and `entity2`

Reply:
But only if the query spaces of `entity1` and `entity2` intersect with the query spaces of `VersionEntity`, right?

Reply:
but if we have 
```
@Entity
class VersionEntity{
 ....
}

@Entity 
class TestEntity{
...
}

session.persist( testEntity1 );
session.persist( testEntity2 );

session.createQuery( "from VersionEntity v")
  .setLockMode( LockModeType.PESSIMISTIC_FORCE_INCREMENT )
  .getResultList();

```
the batch is executed after the translation completes

Reply:
Right, that's what I assumed. So executing the current batch eagerly on the query might change the order of operations. The reason we can't batch the force increment operations is that we can only have a single active batch right now. That's fine though, let's merge this PR and do further optimizations as part of https://hibernate.atlassian.net/browse/HHH-17077